### PR TITLE
Feat: Weaviate delete with filter

### PR DESCRIPTION
**Pull request by neebdev**
Support for deleting objects based on either their ids or a filter condition in weaviate vectorstore.

### CODE REVIEW

File: langchain/src/vectorstores/weaviate.ts [hunk 1]
```diff
@@ -155,13 +155,29 @@ export class WeaviateStore extends VectorStore {
...
-        .deleter()
+  async delete(params: { ids: string[] }): Promise<void>;
+
+  async delete(param: { filter: WeaviateFilter }): Promise<void>;
+
+  async delete(params: {
+    ids?: string[];
+    filter?: WeaviateFilter;
+  }): Promise<void> {
+    const { ids, filter } = params || {};
```
Comment (jacoblee93) [inline langchain/src/vectorstores/weaviate.ts:166]: `params` should always be defined no?

**Comment by jacoblee93** [comment]
One small nit

File: langchain/src/vectorstores/weaviate.ts (hunk 1 above)
Comment (neebdev) [reply langchain/src/vectorstores/weaviate.ts:166]: yeah that or is left redundant. I was first creating it optional for deleting entire index.

Comment (neebdev) [reply langchain/src/vectorstores/weaviate.ts]: I also needed to delete a whole class(index) in weaviate. I was not sure which way was most appropriate. Is it fine to make a static function which takes index name and delete it?

Comment (jacoblee93) [reply langchain/src/vectorstores/weaviate.ts]: Convention at the moment is a `deleteAll` flag like this: https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/vectorstores/pinecone.py#L385

Comment (neebdev) [reply langchain/src/vectorstores/weaviate.ts]: https://weaviate.io/developers/weaviate/manage-data/classes#delete-a-class

Comment (neebdev) [reply langchain/src/vectorstores/weaviate.ts]: If I am not wrong, there is no way to delete all objects while keeping a class in weaviate. You can delete an entire class which is called index in langchain. So implementing a deleteAll which delete the whole index will render the weaviate store instance unusable.

Comment (jacoblee93) [reply langchain/src/vectorstores/weaviate.ts]: Ah ok, sounds a bit more specific. A special method could make sense in this case then, although I assume you'll need auth so a static method might not work well. `deleteClass` perhaps? Or is there a way to do this with a filter?

Comment (neebdev) [reply langchain/src/vectorstores/weaviate.ts]: maybe a static function which takes the client and index name?

Comment (neebdev) [reply langchain/src/vectorstores/weaviate.ts]: another solution would be to delete the class and then create again with same parameter. Doesn't sound right. And someone might need to delete the class and not just the objects. So I guess a static method with client for auth and index name for deletion which returns void is the way to go?

**Comment by neebdev** [comment]
Added the static method. I named it `deleteIndex` for consistency.

**Comment by neebdev** [comment]
Should we add `deleteIndex` as a generic function in vectorstore base because most vector databases has the feature of deleting an index?

**Comment by neebdev** [comment]
@jacoblee93 Let me know if it looks good or it needs changes. Thanks

**Comment by jacoblee93** [comment]
Looking at it now, if this renders the vector store unusable (which it does, right?) without another API call, it seems a bit out of scope for the class. It's a one-liner that someone can do with the client directly anyway.

**Comment by neebdev** [comment]
Good point. It would render an existing index unusable but since it returns void so it can be used to delete a single index with the name. But for now I removed it.

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**